File: TrailFinder.Api/Models/Trail.cs
// Models/Trail.cs
namespace TrailFinder.Api.Models;

public class Trail
{
    public string Id { get; set; } = default!;
    public string Name { get; set; } = default!;
    public string Slug { get; set; } = default!;
    public string Description { get; set; } = default!;
    public decimal DistanceMeters { get; set; }
    public decimal ElevationGainMeters { get; set; }
    public string? DifficultyLevel { get; set; }
    public object? RouteGeom { get; set; }
    public object? StartPoint { get; set; }
    public double? StartPointLatitude { get; set; }
    public double? StartPointLongitude { get; set; }
    public string? WebUrl { get; set; }
    public string? GpxFilePath { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }
    public string UserId { get; set; } = default!;
}

File: TrailFinder.Web/src/types/trail.ts
// src/types/trail.ts
export interface Trail {
    id: string;
    parenId: string;
    name: string;
    slug: string;
    description: string;
    distance_meters: number;
    elevation_gain_meters: number;
    difficulty_level: string | null;
    route_geom: unknown;  // PostGIS type
    start_point: unknown;  // PostGIS type
    start_point_latitude?: number;
    start_point_longitude?: number;
    web_url: string | null;
    has_gpx: boolean;
    created_at: string;
    updated_at: string;
    user_id: string;
}
